File: Sequential Digits.py
# https://leetcode.com/problems/sequential-digits/?envType=daily-question&envId=2024-02-02
# Sequential Digits

class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        # nums = []
        
        # for i in range(1, len(str(high))):
        #     for j in range(1, 10-i):
        #         num = str(j)
        #         for k in range(i):
        #             num += str((j+1)+k)
        #         nums.append(int(num))

        # return [num for num in nums if num >= low and num <= high]

        seq_digits = "123456789"
        result = []

        for length in range(len(str(low)), len(str(high)) + 1):
            for start in range(10 - length):
                num = int(seq_digits[start:start + length])
                if low <= num <= high:
                    result.append(num)

        return result

File: Sequential Digits.js
// https://leetcode.com/problems/sequential-digits/?envType=daily-question&envId=2026-07-13
// Sequential Digits

/**
 * @param {number} low
 * @param {number} high
 * @return {number[]}
 */
var sequentialDigits = function(low, high) {
    
    const seqDigits = "123456789";
    const result = [];

    const minLen = String(low).length;
    const maxLen = String(high).length;

    for (let length = minLen; length <= maxLen; length++) {
        for (let start = 0; start <= 9 - length; start++) {
            const num = Number(seqDigits.slice(start, start + length));

            if (num >= low && num <= high) {
                result.push(num);
            }
        }
    }

    return result;

};